runtime/descriptor.go

```diff
@@ -98,24 +98,62 @@ func propertySet(f *Frame, desc, inst, value *Object) *BaseException {
 
 // makeStructFieldDescriptor creates a descriptor with a getter that returns
 // the field given by fieldName from t's basis structure.
-func makeStructFieldDescriptor(t *Type, fieldName, propertyName string) *Object {
+func makeStructFieldDescriptor(t *Type, fieldName, propertyName string, writtable bool) *Object {
 	field, ok := t.basis.FieldByName(fieldName)
 	if !ok {
 		logFatal(fmt.Sprintf("no such field %q for basis %s", fieldName, nativeTypeName(t.basis)))
 	}
+
 	getterFunc := func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 		var ret *Object
 		var raised *BaseException
-		if raised = checkFunctionArgs(f, fieldName, args, ObjectType); raised == nil {
-			o := args[0]
-			if !o.isInstance(t) {
-				format := "descriptor '%s' for '%s' objects doesn't apply to '%s' objects"
-				raised = f.RaiseType(TypeErrorType, fmt.Sprintf(format, propertyName, t.Name(), o.typ.Name()))
-			} else {
-				ret, raised = WrapNative(f, t.slots.Basis.Fn(o).FieldByIndex(field.Index))
-			}
+
+		if raised = checkFunctionArgs(f, fieldName, args, ObjectType); raised != nil {
+			return ret, raised
 		}
+
+		self := args[0]
+		if !self.isInstance(t) {
+			format := "descriptor '%s' for '%s' objects doesn't apply to '%s' objects"
+			raised = f.RaiseType(TypeErrorType, fmt.Sprintf(format, propertyName, t.Name(), self.typ.Name()))
+			return ret, raised
+		}
+
+		ret, raised = WrapNative(f, t.slots.Basis.Fn(self).FieldByIndex(field.Index))
 		return ret, raised
 	}
-	return newProperty(newBuiltinFunction("_get"+fieldName, getterFunc).ToObject(), None, None).ToObject()
+
+	setterFunc := func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+		ret := None
+		var raised *BaseException
+		if raised = checkFunctionArgs(f, fieldName, args, ObjectType, ObjectType); raised != nil {
+			return ret, raised
```

Comment:
Will the caller handle the ``nil, raised`` instead of the expected ``None, raised``?

I am trying to keep this compatible with the existing version, so am holding myself from external-facing changes.

Can be done. Please confirm.